lib/samly/router_util.ex

```diff
@@ -8,7 +8,9 @@ defmodule Samly.RouterUtil do
 
   @subdomain_re ~r/^(?<subdomain>([^.]+))?\./
 
-  def check_idp_id(conn, _opts) do
+  def check_idp_id(%Conn{private: %{samly_idp: %IdpData{}}} = conn, _opts), do: conn
+
+  def check_idp_id(conn, _opts), do: conn
     idp_id_from = Application.get_env(:samly, :idp_id_from)
```

Comment:
How does this line ever return anything but `nil` ?
`idp_id_from = Application.get_env(:samly, :idp_id_from)`

Reply:
This line exists in the current codebase and is unaffected by this PR.

Reply:
@bmuller ya i was just reviewing your PR and i just happen to be reviewing this method. And that line will never actually return the value its trying to access.

The way it says in the README to setup your configs is for example
```
config :samly, Samly.Provider,
  idp_id_from: :path_segment,
  ```
  
  So you would need to have that line be
  ```
  Application.get_env(:samly, Samly.Provider)[:idp_id_from]
  ```

Reply:
Basically every where in this app it calls Application.get_env() its doing it in conflict with how it says to define it in the README.